vod/hls/m3u8_builder.c

```diff
@@ -1093,9 +1163,13 @@ m3u8_builder_write_variants(
 		{
 			p = vod_sprintf(p, M3U8_STREAM_TAG_SUBTITLES, 0);
 		}
-		else
+		if (media_set->closed_captions_count == 0 && media_set->has_closed_captions)
+		{
+			p = vod_sprintf(p, M3U8_STREAM_TAG_NO_CLOSED_CAPTIONS);
+		}
+		if (media_set->closed_captions_count > 0 && media_set->has_closed_captions)
```

Comment:
redundant, checking count > 0 is enough (note this check repeats multiple times below)